cody.el

```diff
@@ -12,7 +12,8 @@
 ;; Load this package and then add `(cody-start)' to your .emacs
 
 ;;; Code:
-(require 'cl-lib)
+(eval-when-compile (require 'cl-lib))
+(eval-when-compile (require 'eieio-base))
```

Comment:
so to help expand my elisp knowledge, is this a performance optimization? From reading the docs on eval-when-compile it seems that it means only do the work at compile time, not at both compile and eval time. So the reason you do this is because we only use macros from these packages (and these macros don't rely on runtime stuff inside of the package?)

Reply:
Exactly; this is just for the macros, so we don't need them at runtime.